internal/fields/validate.go

```diff
@@ -89,16 +90,25 @@ func CreateValidatorForDataStream(dataStreamRootPath string, opts ...ValidatorOp
 
 	v.allowedCIDRs = initializeAllowedCIDRsList()
 
-	v.Schema, err = loadFieldsForDataStream(dataStreamRootPath)
+	fieldsDir := filepath.Join(fieldsParentDir, "fields")
+	v.Schema, err = loadFieldsFromDir(fieldsDir)
 	if err != nil {
-		return nil, errors.Wrapf(err, "can't load fields for data stream (path: %s)", dataStreamRootPath)
+		return nil, errors.Wrapf(err, "can't load fields for data stream (path: %s)", fieldsParentDir)
 	}
 
 	if v.disabledDependencyManagement {
 		return v, nil
 	}
 
-	packageRoot := filepath.Dir(filepath.Dir(dataStreamRootPath))
+	packageRoot, found, err := packages.FindPackageRoot()
+	if err != nil {
+		return nil, errors.Wrap(err, "can't find package root")
+	}
+	if !found {
```

Comment:
Could this hide problems with malformed packages?

On what cases is it ok to not find the package root? Could you add a comment here about this?

Reply:
I added a comment, let me know what do you think.

Reply:
Ok, we can revisit this if found to be problematic at some point. Would it make sense to add a debug message here?

Reply:
Sure!